libqpdf/QPDF_linearization.cc

```diff
@@ -325,11 +325,10 @@ QPDF::readLinearizationData()
 QPDFObjectHandle
 QPDF::readHintStream(Pipeline& pl, qpdf_offset_t offset, size_t length)
 {
-    int obj;
-    int gen;
+    QPDFObjGen og;
     QPDFObjectHandle H = readObjectAtOffset(
-        false, offset, "linearization hint stream", -1, 0, obj, gen);
-    ObjCache& oc = this->m->obj_cache[QPDFObjGen(obj, gen)];
+        false, offset, "linearization hint stream", QPDFObjGen(-1), og);
```

Comment:
After my upcoming commit, this will need to be an explicit `QPDFObjGen(0, 0)`. I'd rather have the explicit `0` than rely on the default -- see comments in `QPDF.cc` after my commit.